File: spring-web-java/src/main/java/spring/web/java/domain/model/entity/Notification.java
package spring.web.java.domain.model.entity;

import jakarta.persistence.Column;
import jakarta.persistence.ConstraintMode;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.ForeignKey;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Notification extends Base {

	@Id
	@GeneratedValue
	@Column(name = "notification_id")
	private Long id;
	private String title;
	private String message;
	private String url;
	private boolean isRead = false;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "member_id", foreignKey = @ForeignKey(ConstraintMode.NO_CONSTRAINT))
	private Member member;

	public static Notification of(Member member, String title, String message, String url) {
		Notification notification = new Notification();

		notification.member = member;
		notification.title = title;
		notification.message = message;
		notification.url = url;
		notification.isRead = false;

		return notification;
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/model/entity/Notification.kt
package spring.web.kotlin.domain.model.entity

import jakarta.persistence.*

@Entity
class Notification protected constructor(
    val title: String,
    val message: String,
    val url: String,
    isRead: Boolean,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", foreignKey = ForeignKey(ConstraintMode.NO_CONSTRAINT))
    var member: Member,
) : Base() {
    @Id
    @GeneratedValue
    @Column(name = "notification_id")
    var id: Long? = null
        protected set

    var isRead: Boolean = isRead
        protected set

    companion object {
        fun of(member: Member, title: String, message: String, url: String) =
            Notification(
                title = title,
                message = message,
                url = url,
                isRead = false,
                member = member
            )
    }
}